### PR TITLE
chore: use Rust 1.65.0

### PR DESCRIPTION
**Pull request by AaronO**
Since it released as stable 2 weeks ago, there's a considerable amount of minor lints ...

### CODE REVIEW

File: cli/tools/coverage/range_tree.rs
Comment (dsherret) [inline cli/tools/coverage/range_tree.rs:75]: I don't understand this change. Not a big deal, but do you know why?

**Review by dsherret — APPROVED** [review]
LGTM! Thanks @AaronO and hope you're doing well!

File: cli/tools/coverage/range_tree.rs
Comment (AaronO) [reply cli/tools/coverage/range_tree.rs:75]:
```rust
error: parameter is only used in recursion
   --> cli/tools/coverage/range_tree.rs:75:5
    |
75  |     rta: &'a RangeTreeArena<'a>,
    |     ^^^ help: if this is intentional, prefix it with an underscore: `_rta`
    |
    = note: `-D clippy::only-used-in-recursion` implied by `-D clippy::all`
note: parameter used here
   --> cli/tools/coverage/range_tree.rs:99:46
    |
99  |           children.push(RangeTree::normalize(rta, head));
    |                                              ^^^
...
113 |         children.push(RangeTree::normalize(rta, head));
    |                                            ^^^
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#only_used_in_recursion
```

Comment (AaronO) [reply cli/tools/coverage/range_tree.rs:75]: Happy to remove it in this PR or in a follow-up. Wanted to limit unrelated changes in a toolchain upgrade

Comment (AaronO) [reply cli/tools/coverage/range_tree.rs]: Dropped it in the latest commit, feel free to squash & merge once green (I don't have merge permissions)

**Pull request merged**
